edgegrid/auth.go

```diff
@@ -23,6 +26,9 @@ type AuthParams struct {
 	headersToSign []string
 }
 
+// NewAuthParams is a convenience function for generating
+// an AuthParams{} from an *http.Request, acccess token string, client
+// token string, and a client secret string
 func NewAuthParams(req *http.Request, accessToken, clientToken, clientSecret string) AuthParams {
```

Comment:
Stylistic comment: It is not typical to annotate type names with `{}` in documentation comments, and it is more idiomatic to refer to function arguments by their name in the function signature. This doc comment would be more idiomatic written like this:

```go
// NewAuthParams returns an AuthParams generated from req, accessToken,
// clientToken, and clientSecret.
```

There is no need to mention the types of the function arguments because the reader can see them in the source code directly below the comment and tools like godoc and godoc.org always show the function signature next to the documentation.